### PR TITLE
added unit tests for AEPPushNotificationBuilder

### PR DESCRIPTION
**Pull request by PratthamArora**
## Description
Added unit tests for AEPPushNotificationBuilder

## Related Issue
https://jira.corp.adobe.com/browse/MOB-21012

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    private fun verifyNotificationDataFields(
+        notification: Notification,
+        pushTemplate: AEPPushTemplate
+    ) {
+        val pendingIntent = notification.contentIntent
+        val shadowPendingIntent = Shadows.shadowOf(pendingIntent)
+        val intent = shadowPendingIntent.savedIntent
+
+        assertEquals(Notification::class.java, notification.javaClass)
+        assertEquals(pushTemplate.ticker, notification.tickerText)
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:185]: nit: can we also verify the channel id, badge count, visibility, sound and priority settings here, if possible?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            smallLayout,
+            expandedLayout,
+            CONTAINER_LAYOUT_VIEW_ID
+        ).build()
+
+        verifyNotificationDataFields(notification, pushTemplate)
+    }
+
+    @Test
+    fun `verify construct should map valid data fields and colors to notification view`() {
```
Comment (spoorthipujariadobe) [inline code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:90]: nit: Any reason not to club this test and the one before this?

**Review by spoorthipujariadobe — APPROVED** [review]

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt (hunk 2 above)
Comment (PratthamArora) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:90]: no reason just wanted to keep notification-related tests and notification view-related tests separate, if you want I can combine them into one, what do you suggest?

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt (hunk 1 above)
Comment (PratthamArora) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:185]: sure will add them

File: code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt (hunk 2 above)
Comment (spoorthipujariadobe) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:90]: Don't really have a preference. Noticed that it was combined for the subsequent test that uses intent. No issue with this :)

Comment (PratthamArora) [reply code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/AEPPushNotificationBuilderTest.kt:90]: Yeah, that was intentional, I wanted to keep view related tests separate for better readability, however once it was validated for map then it made sense to combine it for the intent method

will proceed with this pattern

**Pull request merged**